File: LeetCode/JavaScript/#434 Number of Segments in a String.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
   s = s.trim();
   if(s===""){
     return 0;
   }
   var res = 0;
   res = s.replace(/\s+/g," ").split(" ").length;
   return res;
};

File: LeetCode/Python/#434 Number of Segments in a String.py
class Solution(object):
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        return len(s.split())



class Solution(object):
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        flag = True

        for i in s:
            if i!="":
                res += flag
                flag = False
            else:
                flag = True

        return res
